src/main/java/ch/njol/skript/structures/StructVariables.java

```diff
@@ -134,33 +214,42 @@ public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResu
 					continue;
 				}
 			}
-
 			variables.add(new NonNullPair<>(name, o));
 		}
+		getParser().getCurrentScript().addData(new DefaultVariables(variables));
 		return true;
 	}
 
 	@Override
 	public boolean load() {
-		for (NonNullPair<String, Object> pair : variables) {
+		DefaultVariables data = getParser().getCurrentScript().getData(DefaultVariables.class);
+		for (NonNullPair<String, Object> pair : data.getVariables()) {
 			String name = pair.getKey();
-			Object o = pair.getValue();
-
 			if (Variables.getVariable(name, null, false) != null)
 				continue;
 
-			Variables.setVariable(name, o, null, false);
+			Object value = pair.getValue();
+			Variables.setVariable(name, value, null, false);
```

Comment:
```suggestion
			Variables.setVariable(name, pair.getValue(), null, false);
```
We can avoid initializing a variable here